ReactiveUI.Fody/ReactiveUIPropertyWeaver.cs

```diff
@@ -54,6 +58,59 @@ public void Execute()
             {
                 foreach (var property in targetType.Properties.Where(x => x.IsDefined(reactiveAttribute)).ToArray())
                 {
+                    TypeReference genericTargetType = targetType;
+                    if (targetType.HasGenericParameters)
+                    {
+                        var genericDeclaration = new GenericInstanceType(targetType);
+                        foreach (var parameter in targetType.GenericParameters)
+                        {
+                            genericDeclaration.GenericArguments.Add(parameter);
+                        }
+                        genericTargetType = genericDeclaration;
+                    }
+
+                    if (property.SetMethod == null && property.GetMethod.IsExpressionBodiedProperty())
+                    {
+                        // Add this.RaisePropertyChange(PropertyName) to the setter of any properties referenced 
+                        // by this expression bodied property where they have an available setter
+                        var getInstructions = property.GetMethod.Body.Instructions.Where(i =>
+                        {
+                            var operand = i.Operand as MethodDefinition;
+                            return i.OpCode == OpCodes.Call && (operand?.IsGetter ?? false);
+                        });
+
+                        var setMethodsForGetInstructions =
+                            targetType.Properties.Where(
+                                    prop =>
+                                        prop.SetMethod != null &&
+                                        prop.SetMethod.HasBody &&
+                                        getInstructions.Any(
+                                            i => ((MethodDefinition) i.Operand).FullName == prop.GetMethod.FullName))
+                                .Select(prop => prop.SetMethod);
```

Comment:
Per the above comment, this could then be simplified to:

```
var setMethodsForGetInstructions = methodDependencies
    .Where(x => setMethodsByGetMethod.ContainsKey(x => x))
    .Select(x => setMethodsByGetMethod[x])
    .ToArray();
```

(The `.ToArray()` to avoid multiple traversals of the enumerable)